packages/core/src/errors.ts

```diff
@@ -427,3 +441,21 @@ export class Verification extends AuthError {
 export class MissingCSRF extends SignInError {
   static type = "MissingCSRF"
 }
+
+// TODO: Review this list to make sure it's complete and we only send non-hinting errors to the client.
+const clientErrors = new Set<ErrorType>([
+  "CredentialsSignin",
+  "OAuthAccountNotLinked",
+  "OAuthCallbackError",
+  "OAuthSignInError",
+  "EmailSignInError",
+])
```

Comment:
It's important that we go through this list and include all non-sensitive error types here.